js/data/bucket.js

```diff
@@ -109,6 +109,35 @@ Bucket.prototype.populateArrays = function() {
 };
 
 /**
+ * Update paint arrays with the given feature properties, leaving geometries
+ * as-is.
+ * @private
+ */
+Bucket.prototype.updatePaintVertexArrays = function(programName, propertiesList) {
+    this.recalculateStyleLayers();
+
+    for (var i = 0; i < propertiesList.length; i++) {
+        var range = this._featureIndexToArrayRange[i];
+        if (!range) continue;
+        var numVertices = range.endVertex - range.startVertex + (range.endGroup - range.startGroup) * ArrayGroup.MAX_VERTEX_ARRAY_LENGTH;
+        this.prepareArrayGroup(programName, numVertices);
+        this.populatePaintArrays(programName, {zoom: this.zoom}, propertiesList[i], range);
+
+        // Maintain the ArrayGroup invariant that layoutVertexArray.length ===
+        // paintVertexArrays[*].length, necessary for proper operation of
+        // prepareArrayGroup
+        for (var g = range.startGroup; g <= range.endGroup; g++) {
+            var start = g === range.startGroup ? range.startVertex : 0;
+            var end = g === range.endGroup ? range.endVertex : 0;
+            for (var v = start; v <= end; v++) {
+                this.arrayGroups[programName][g].layoutVertexArray.emplaceBack(0, 0);
+            }
+        }
```

Comment:
@lucaswoj Without this code, the invariant in question would not be met here because this method, `updatePaintVertexArrays`, would then only be resizing and populating the _paint_ arrays, and not the _layout_ arrays.    (The layout arrays are otherwise only sized/populated during `addFeature`.)

The only reason this invariant even matters here, however, is because it is assumed by the `prepareArrayGroup` discussed above.  The right refactor there could eliminate the need to do this here.